filebeat/module/haproxy/log/ingest/pipeline.json

```diff
@@ -29,6 +29,26 @@
                 ]
             }
         },
+        {
+            "date": {
+                "formats": [
+                    "dd/MMM/yyyy:HH:mm:ss.SSS",
+                    "MMM dd HH:mm:ss"
+                ],
+                "timezone" : "{{ event.timezone }}",
```

Comment:
Maybe we should check `"if": "ctx.event.timezone == null"` first? Similar to https://github.com/elastic/beats/blob/master/filebeat/module/system/syslog/ingest/pipeline.json#L30-L58